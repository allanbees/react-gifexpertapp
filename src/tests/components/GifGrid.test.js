import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Tests for <GifGrid />', () => {
    const category = 'Programming';
    
    test('Must render the component properly', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category = { category } />);
        expect( wrapper ).toMatchSnapshot();

    });

    test('Must display items when gifs are loaded with useFetchGifs', () => {

        const gifs = [{
            id: 'ABC', 
            url: 'https://reactjs.org/',
            title: 'title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category = { category } />);
        
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });

});
