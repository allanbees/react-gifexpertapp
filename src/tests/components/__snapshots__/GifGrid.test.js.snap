// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGrid /> Must render the component properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Programming
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading data...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
